### ISSUE TITLE
Yolov5 model detects everything as cars

### ISSUE THREAD

**Issue by Pedro-Leitek** [ultralytics/yolov5#10353]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi,

I'm facing an issue regarding my yolov5 model to detect cars.
Here is the following procedure I made to train the model:
-Downloaded 10000 images from training car dataset.
-Imported them to roboflow
- Splited the images so the training dataset is 7500 images and validation dataset 2500 images.
- Exported it as yolov5 pytorch.
- Trained the model as img size 640, 100 epochs, device GPU, model yolov5m.pt
- Exported the model to onnx

If I put an image of cars in the model, the cars are detected, however lots of false-positives happen when using images without cars. For instance I have a camara in the place I work + a python program for processing. The idea is to make the camera rotate, capture a snapshot, processing and rotate again. Lets say the camera points to a wall or cabinet or computer and it detects everything as cars. I noticed that the detection box, in lots of  the frames, cover the full frame in Full HD and a quarter in a UHD image (top left).
I would like to know what are the possible causes of this behaviour.

Thanks

**Comment by JustasBart** [comment]
Hi, I can give you some pointer that might hopefully help you out:

First of all what is the input size of your images? Because if you for example have a feed of 1920x1080 you are then resizing that frame to 640x640 which might squeeze it by so much that it just can't see the cars clearly - and thus everything is a car!
- To fix this consider training with the --rect and set the input size to maybe 1280 or whatever you can afford in terms of training such so that your frame will not have to be squished to a square, but will also be bigger. (You can go all the way to 1920x1088 though that would make it rather expensive computationally for both training and inference).

The other thing to consider is the scale of what you've trained your model on and what your camera is seeing. This is because ideally you should train your model on the data that you expect for it to look for, but in your case it sounds like you're expecting it to detect stuff 'In the wild' while your data is whatever you were able to find elsewhere.
- To fix this try and make sure that the training/validation data matches as closely to your actual inference data as possible.

Another pointer is that you should include some negative data of your actual environment, that is show it unlabelled images of all the things that you don't want it to detect so that it trains in a way where it's hopefully not picking that up.

The last few things is make sure that you train it with enough epochs say even 1,000 epochs as it has a patience flag of 100 (By default) meaning that it won't let you to over-train that easily anyway, but in this case more epochs could be a good thing depending on the data.

And lastly just keep on experimenting! I would recommend to start off with like 200 images train and 60 val (Or thereabouts) to keep your train/test iterations quick and just see what works and what doesn't!

I hope any of that is of any use to you, good luck!

**Comment by Pedro-Leitek** [comment]
@JustasBart Regarding input size, in roboflow there is an option to resize the the image (I put it as 640x640).
In my script to control the camera and processing I resize the full hd image to 640x640 also.
The camera is to be installed outside office in the future, but I find weird that it detects everything as car in the office. I mean each frame processed is a car as the bounding box covers the whole frame. I think the negative data is not the issue here. You see I have 3 more models of objects and the FP are low, compared to the model of cars. The other models where trained without negative data. All the models where training following the same criteria (img size 640, yolov5m, dataset split 75% train 25% validation). What baffles me is that the object is detected in the full frame if the image is full hd but a quarter of the frame (top left) in 4K. I saw the images of the and they seem completely normal. Any idea why the full hd image has a bounding box different from a 4K?

Thanks

**Comment by JustasBart** [comment]
@Pedro-Leitek I see, okay now I'm on board with you that yeah that does seem weird! I thought that maybe it was just a generic issue sort of a way, but it sounds like there might be more to it!

I'm afraid I'm of no use to you when it comes to this but I hope you can get this resolved sometime soon enough!

All the best!

**Comment by Pedro-Leitek** [comment]
Hi, I would like more help regarding this issue.

Thanks

**Comment by YeopSeung04** [comment]
Hi.

I had a similar experience with you.
I also took a soccer ball data set from roboflow and learned it.
However, all similar things such as human faces, bicycle wheels, etc. were detected with soccer balls.

I found two problems.

The first question
My learning image resolution was Full HD like yours.
The resolution of the image predicting the result was 4K.

Come to think of it, it is different from full HD soccer balls and 4K soccer balls in pixels.

Yolo learns by calculating the image in pixel units in a convolutional CNN scheme.

In other words, if you learned with Full HD, it is better to use a Full HD prediction image.

The second question.
It was not enough learning.

I thought that if there was a lot of images, I would be able to learn at least
Yolo knew it was a model for performance instead of accuracy.

It would be nice to increase the number of lessons from 100.

I want you to change your way of learning.

If you learn again with --img 416 --batch 16 --epochs 100, you'll get good results.

**Comment by glenn-jocher** [comment]
@AI-Expert-04 hi,

Thank you for sharing your experience. It sounds like you encountered similar issues with object detection.

Regarding image resolution, using consistent image sizes for training and prediction is definitely important for accurate detection. Additionally, increasing the number of epochs and adjusting the training parameters like --img and --batch size might help improve your model's performance.

Try training your model with --img 640 (since you've used 640x640 images) and more epochs to see if it improves the detection accuracy. It's also useful to regularly monitor the training process and adjust parameters if needed.

Hope this helps! Let me know if you have any more questions.
